src/main/index.ts

```diff
@@ -146,9 +158,19 @@ if (!isFirstInstance) {
   };
 
   app.on('session-created', (session) => {
+    let reservedStateValues = [];
     if (!isDevelopment) {
       log.info('Cleaning local storage on session creation');
-      session.clearStorageData({ storages: ['localstorage'] }).catch(log.error);
+
+      if (mainWindow != null) {
+        mainWindow.webContents.executeJavaScript('({...localStorage});', true).then((localStorage) => {
+          reservedStateValues = reservedStates.map((state) => localStorage.getItem(state));
+          session.clearStorageData({ storages: ['localstorage'] }).catch(log.error);
+          for (let i = 0; i < reservedStates.length; i++) {
+            localStorage.setItem(reservedStates[i], reservedStateValues[i]);
```

Comment:
This is not the `localStorage` object in the renderer process, I don't think `.setItem()` exists - have you verified this code works?